app/src/main/java/com/eveningoutpost/dexdrip/Models/Calibration.java

```diff
@@ -906,9 +912,15 @@ public static void adjustRecentBgReadings(int adjustCount) {
                             bgReading.filtered_calculated_value = new_calculated_value;
                         }
                         bgReading.calculated_value = new_calculated_value;
+                        
+                        Log.v(TAG, "Previous value: " + oldYValue + ", New value: " + bgReading.calculated_value);
 
                         bgReading.save();
                         BgReading.pushBgReadingSyncToWatch(bgReading, false);
+                        if (uploadModified) {
+                            UploaderQueue.newEntry("update", bgReading);
+                            SyncService.startSyncService(3000); // sync in 3 seconds
```

Comment:
Do we need to sync in every step of the iteration? What do you think about doing a single sync after we are done iterating over `bgReadings`

Reply:
Yea, there's no fantastic reason for why I did this, other than it mimics the watch syncing behaviour. I'll revisit it and almost certainly make the change.

Reply:
Changes made, still need to test - will be building shortly and will start running immediately.